index.js

```diff
@@ -54,31 +54,39 @@ function summary (scanner) {
     node.children.push(t)
   }
 
-  if (scanner.peek() === ':' || isSummarySep(scanner.peek())) {
-    // <type> <summary-sep> <text>
-    node.children.push(summarySep(scanner))
-    node.children.push(text(scanner))
-  } else if (scanner.peek() === '(') {
-    // <type> "(" <scope> ")" <summary-sep> <text>
+  let s
+  if (scanner.peek() === '(') {
+    // <type> "(" <scope> ")" ...
     scanner.next()
-    const s = scope(scanner)
+    s = scope(scanner)
     if (s instanceof Error) {
       return s
     } else {
       node.children.push(s)
     }
-    if (scanner.peek() !== ')') return invalidToken(scanner, [')'])
-    scanner.next()
-    const sep = summarySep(scanner)
-    if (sep instanceof Error) {
-      return sep
-    } else {
-      node.children.push(sep)
+    if (scanner.peek() !== ')') {
+      return invalidToken(scanner, [')'])
     }
-    node.children.push(text(scanner))
+    scanner.next()
+  }
+
+  // ... ["!"] ...
+  let b = breakingChange(scanner)
+  if (b instanceof Error) {
+    b = null
+  } else {
+    node.children.push(b)
+  }
+
+  // ... ": " <text>
+  const sep = separator(scanner)
+  if (sep instanceof Error) {
+    return invalidToken(scanner, [!s && '(', !b && '!', ':'])
```

Comment:
This is a quick way to enable/disable certain "allowed characters" based on previously parsed data. E.g.

- if a scope was defined, we should skip the `(` character.
- if the short-breaking change notion is used, skip the `!` character

Reply:
Seems smart to me.